src/component/providers/UIProvider/UIProvider.tsx

```diff
@@ -1,19 +1,47 @@
 import React, { useState } from 'react';
-import UIContext, { createEmptyToast } from 'contexts/UIContext';
+import UIContext, { createEmptyToast, themeMode } from 'contexts/UIContext';
 import { IToast } from 'interfaces/toast';
+import { getLocalStorageItem } from 'utils/storage';
 
-const UIProvider: React.FC = ({ children }) => {
+const resolveMode = (darkmode: boolean): themeMode => {
+    const value = getLocalStorageItem('unleash-theme');
+    if (value) {
+        return value as themeMode;
+    }
+
+    let osDark;
+    if (darkmode) {
+        osDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
```

Comment:
Would be cool if this was a custom `useEffect` hook that automatically updates on OS dark/light mode change.

* https://usehooks.com/useDarkMode/
* https://usehooks.com/useMedia/